AppCenterDataStorage/AppCenterDataStorage/MSDataStore.h

```diff
@@ -0,0 +1,132 @@
+#import "MSSerializableDocument.h"
+#import "MSServiceAbstract.h"
+
+@class MSDocumentWrapper<T : id<MSSerializableDocument>>;
+@class MSPaginatedDocuments<T : id<MSSerializableDocument>>;
+@class MSReadOptions;
+@class MSWriteOptions;
+@class MSDataStoreError;
+
+/**
+ * App Data Storage service.
+ */
+
+NS_ASSUME_NONNULL_BEGIN
+
+// User documents
+// An authenticated user can read/write documents in this partition
+static NSString *const MSDataStoreUserDocumentsPartition = @"user-{userid}";
+
+// Application partition
+// Everyone can read documents in this partition
+// Writes is not allowed via the SDK
+static NSString *const MSDataStoreAppDocumentsPartition = @"readonly";
+
+/**
+ * Time to live constants
+ */
+static int const MSDataStoreTimeToLiveInfinite = -1;
+static int const MSDataStoreTimeToLiveNoCache = 0;
+static int const MSDataStoreTimeToLiveDefault = 60 * 60;
+
+@interface MSDataStore<T : id <MSSerializableDocument>> : MSServiceAbstract
+
+typedef void (^MSDocumentWrapperCompletionHandler)(MSDocumentWrapper<T> *document);
+typedef void (^MSPaginatedDocumentsCompletionHandler)(MSPaginatedDocuments<T> *documents);
+typedef void (^MSDataStoreErrorCompletionHandler)(MSDataStoreError *error);
+
+/**
+ * Change The URL that will be used for getting token.
+ *
+ * @param tokenExchangeUrl The new URL.
+ */
++ (void)setTokenExchangeUrl:(NSString *)tokenExchangeUrl;
+
+/**
```

Comment:
applies to all functions: please update the method description and provide params and descriptions

Reply:
Got it, updated.